java-frontend/src/main/java/org/sonar/java/classpath/AbstractClasspath.java

```diff
@@ -130,6 +131,9 @@ static void logResolvedFiles(String property, Collection<File> files) {
         }
         validateLibraries = validateLibs;
         result.addAll(libraryFilesForPattern);
+        if (pathPattern.endsWith("android.jar")) {
+          inAndroidContext = true;
+        }
```

Comment:
The important lines of the first commit are here, all the rest is just passing the value around and adding the possibility to test in unit tests.